src/contract_client/client.ts

```diff
@@ -45,6 +47,35 @@ export class ContractClient {
     });
   }
 
+  /**
+   * Generate a ContractClient instance from the ContractClientOptions and the wasm binary
+   */
+  static async fromWasm(options: ContractClientOptions, wasm: BufferSource): Promise<ContractClient> {
+    const wasmModule = await WebAssembly.compile(wasm);
+    const xdrSections = WebAssembly.Module.customSections(wasmModule, "contractspecv0");
+    if (xdrSections.length === 0) {
+      return Promise.reject(new Error('Could not obtain contract spec from wasm'));
```

Comment:
I haven't seen this "in case of error within `async` function, return a `Promise.reject`" pattern before. Is this codebase doing this elsewhere? I've most commonly just seen a `throw new Error` instead. Which also seems to be what this StackOverflow answer recommends: https://stackoverflow.com/a/42453705

Reply:
@Shaptic do you have opinions on `return Promise.reject(new Error(...))` vs `throw new Error(...)`?

Reply:
changed to throw Error instead

Reply:
no opinions tbh, but throw seems more readable to a non-`Promise`r

Reply:
So I ended up going back to promise.reject because it was convenient for that test case in the mocha test file which I patterned after the `get_contract_data_test.js`. I noticed that server.ts was returning a promise.reject when it wasn't getting entries back from the server that it expected, so it was after an await call whereas it was throwing errors for stuff like malformed user input (TypeError for example). So based on that pattern it made sense for me to return a promise.reject with error code 404.